### PR TITLE
object_hata OK

### PR DESCRIPTION
**Pull request by EllipticEllipsis**
**N.B.** This object crashes the current version of z64utils due to faulty detection of the collision at the beginning. I hear that @Thar0 has implemented a fix, but it's not in yet.

Debinarised object_hata completely, added symbols to EnHata, made an enum for limbs, renamed a couple of things.

Unfortunately, the way the limbs are used in code implied it was necessary to make an enum, but the structure of the skeleton in this one made the limbs annoying to name: essentially it has the shape
```
0
    1*
        2
            3
                4
                    5
                        6
                            7*
                        8*
                    9*
                10*
        11
            12
                13
                    14
                        15
                            16*
                        17*
                    18*
                19*
```
where `*` means a dList is attatched. Each flag is formed from nodes where the first child is the next position, the second has the other limb, not displaced, with the dList attached. (This structure may be typical of non-flex skeletons, I'm not sure.) I went with naming the limbs with dLists after the dList, and the corresponding parent with the suffix `Base`; I'm open to better alternatives.

Also noticed that the unused data bore a striking resemblence to a `ColliderCylinderInit` and `CollisionCheckInfoInit2`, Combined with the suspicious 0x4C-sized gap in the struct, I took it as sufficient evidence to format it as a ColliderInit and write the gap as a collider, with a note to say it's unused.

Finally, fixed a couple of compiler warnings in `EnRiverSound`.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]
Looks nice and clean.

File: src/overlays/actors/ovl_En_Hata/z_en_hata.c [hunk 1]
```diff
@@ -71,41 +83,48 @@ void EnHata_Update(Actor* thisx, GlobalContext* globalCtx2) {
     GlobalContext* globalCtx = globalCtx2;
     EnHata* this = THIS;
     s32 pitch;
-    Vec3f sp48 = sVec;
-    Vec3f sp3C;
+    Vec3f sZeroVec = { 0.0f, 0.0f, 0.0f };
+    Vec3f sWindVec;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Hata/z_en_hata.c:87]: Ideally the `s` prefix scheme should be kept for static variables
```suggestion
    Vec3f zeroVec = { 0.0f, 0.0f, 0.0f };
    Vec3f windVec;
```

File: src/overlays/actors/ovl_En_Hata/z_en_hata.c [hunk 2]
```diff
@@ -71,41 +83,48 @@ void EnHata_Update(Actor* thisx, GlobalContext* globalCtx2) {
...
+    this->limbs[FLAGPOLE_LIMB_FLAG_2_HOIST_END_BASE].z = this->limbs[FLAGPOLE_LIMB_FLAG_1_HOIST_END_BASE].z;
     this->skelAnime.playSpeed = (Rand_ZeroFloat(1.25f) + 2.75f) * (globalCtx->envCtx.unk_B0 / 255.0f);
 }
 
 s32 EnHata_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
     EnHata* this = THIS;
     Vec3s* limbs;
 
-    if (limbIndex == 12 || limbIndex == 3 || limbIndex == 13 || limbIndex == 4) {
+    if (limbIndex == FLAGPOLE_LIMB_FLAG_2_BASE || limbIndex == FLAGPOLE_LIMB_FLAG_1_BASE || limbIndex == FLAGPOLE_LIMB_FLAG_2_HOIST_END_BASE || limbIndex == FLAGPOLE_LIMB_FLAG_1_HOIST_END_BASE) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Hata/z_en_hata.c:127]: It looks like this file could use a run of the formatter

**Review by Roman971 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Hata/z_en_hata.h [hunk 3]
```diff
@@ -4,11 +4,37 @@
...
+    /* 0x0E */ FLAGPOLE_LIMB_FLAG_2_HOIST_MID_BASE,
+    /* 0x0F */ FLAGPOLE_LIMB_FLAG_2_FLY_MID_BASE,
+    /* 0x10 */ FLAGPOLE_LIMB_FLAG_2_FLY_END_BASE,
+    /* 0x11 */ FLAGPOLE_LIMB_FLAG_2_FLY_END,
+    /* 0x12 */ FLAGPOLE_LIMB_FLAG_2_FLY_MID,
+    /* 0x13 */ FLAGPOLE_LIMB_FLAG_2_HOIST_MID,
+    /* 0x14 */ FLAGPOLE_LIMB_FLAG_2_HOIST_END,
+    /* 0x15 */ FLAGPOLE_LIMB_MAX
+} EnNbLimb;
+
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Hata/z_en_hata.h:32]: enum was copied from nabooru? should change name

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Hata/z_en_hata.h]: Whoops. Fixed

**Review by fig02 — APPROVED** [review]

**Pull request merged**